Add FF android webshare support

hamishwillee opened the pull request:
Fixes #8833

#8833 points out that the web share API works on FF for Android but BCD indicates not supported. 

What I have done is:
- Verified that Navigator.share does indeed work on current FF for android using https://mdn.github.io/dom-examples/web-share/
- Verified that Navigator.canShare works using the [webshare tests](http://www1.www.wpt.live:8000/web-share/canShare-files.tentative.https.html)
- Verified custom gecko integration added in [Bugzilla 1402369](https://bugzilla.mozilla.org/show_bug.cgi?id=1402369) - you can see [code here](https://searchfox.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoSession.java#5062)

Bugzilla indicates this going in in FF72, but there is no FF72 for android so I updated BCD with the first version after that - FF79. 

Note, https://github.com/mdn/browser-compat-data/issues/4983 was created to track these sorts of items.

saschanaz commented on api/Navigator.json:
```diff
@@ -2506,7 +2506,8 @@
             "firefox_android": {
-              "version_added": false
+              "version_added": "79",
+              "notes": "Firefox Mobile displays custom share dialog (instead of the platform-native dialog indicated by the specification)."
```
AFAIK it's using Android feature to customize the UI, which should be still considered as "native". Chrome also shows customized UI and also several non-browser apps e.g. OneDrive.

A quick search shows how. https://stackoverflow.com/q/32478266

saschanaz replied:
I don't see a spec requirement to be "native" BTW. The spec says the user agent "can" use the OS share system, not "must". https://w3c.github.io/web-share/#examples-of-share-targets

hamishwillee replied:
Fair enough. In the abstract it even says "The available share targets are not specified here; they are provided by the user agent.",

I will remove that.

hamishwillee commented:
Thanks @saschanaz . Fixed as you suggested. Change now relatively simple.

saschanaz commented:
Cool! Could you mark it as partial implementation as Firefox doesn't support `files` yet? https://bugzilla.mozilla.org/show_bug.cgi?id=1635700

hamishwillee commented:
@saschanaz Thanks. I added this to the `share` value. Not to `canShare` as my assumption is that will return false to anything that it does not know about. 

Just FMI if I had added supported for web share and THEN files support was added to the spec, would it be expected we would go through and mark all the APIs as "partial"? Seems a little odd.

saschanaz commented:
> 
> 
> @saschanaz Thanks. I added this to the `share` value. Not to `canShare` as my assumption is that will return false to anything that it does not know about.
> 
> Just FMI if I had added supported for web share and THEN files support was added to the spec, would it be expected we would go through and mark all the APIs as "partial"? Seems a little odd.

Fair enough, maybe we should add a subitem instead, something like `file_support`.

saschanaz commented on api/Navigator.json:
```diff
@@ -445,7 +445,7 @@
             },
             "firefox_android": {
-              "version_added": false
+              "version_added": "79"
```
Did you confirm it? Because we haven't added support for `navigator.canShare`.

hamishwillee replied:
Hmmm. I ran http://wpt.live/web-share/canShare-files.tentative.https.html test on my phone with latest FF and it showed success with a single file. Given that we know files are not supported, this is perhaps a false positive.

I will revert that.

hamishwillee replied:
Reverted.

saschanaz replied:
That test is a little confusing. It tests whether passing a file without wrapping it with an array fails or not, and it does fail because it's not supported at all anyway. 😄

hamishwillee replied:
So much for being "clever" in my due diligence. In any case, this is now the most trivial review for you ever!

ddbeck reviewed (approved):
Looks good. Thank you! 🎉
